src/Components/MainComponent/MainComponent.js

```diff
@@ -15,68 +13,88 @@ import {GoogleMapComponent} from '../GoogleMapComponent/GoogleMapComponent'
 import {allCountriesHTTPAddress} from '../../const/const';
 
 export class MainComponent extends Component {
-        constructor(props) {
-            super(props)
-            this.state = {
-                allCountries: [],
-                selectedCountry: {
-                    name: '',
-                    capital: '',
-                    alpha3Code: '',
-                    borders: [],
-                    latlng: [-30, -30]
-                },
-            }
+    constructor(props) {
+        super(props)
+        this.state = {
+            allCountries: [],
+            selectedCountry: {
+                name: '',
+                capital: '',
+                borders: [],
+                latlng: [-30, -30]
+            },
         }
+    }
 
-        componentDidMount = () => {
-            getDataFromAPI(allCountriesHTTPAddress).then((data) => {
-                this.setState({allCountries: data})
-            })
-        };
+    componentDidMount = () => {
+        getDataFromAPI(allCountriesHTTPAddress).then((data) => {
+            this.setState({allCountries: data})
+        })
+    };
+
+    changeCountryButtonClick = (clickedCountry) =>{
+       this.handleUpdateInput(clickedCountry);
+    };
 
-        changedSelectOption = (e) => {
-            e.preventDefault();
-            const alpha3Code = e.target.value;
-            const {allCountries} = this.state;
+    handleUpdateInput = (value) => {
+        const {allCountries} = this.state;
 
-            const selectedCountry = allCountries.find(country => country.alpha3Code === alpha3Code);
+        const selectedCountry = allCountries.find(country => country.name === value);
+        if (selectedCountry != null) {
             const selectedCountryBorders = allCountries.filter((country) =>
                 selectedCountry.borders.find(border => border === country.alpha3Code)
             ).map(country => country.name);
 
             this.setState({
                 selectedCountry: {
                     ...this.state.selectedCountry,
-                    alpha3Code: alpha3Code,
                     name: selectedCountry.name,
                     capital: selectedCountry.capital,
                     borders: selectedCountryBorders,
                     latlng: selectedCountry.latlng
                 }
             })
-        };
+        }
+    };
+
+
+    detailsToDisplay = () => {
+        return (
+            <div>
+                <Paper className='paper'>
+                    <SelectedCountryDetailsComponent
+                        details={this.state.selectedCountry}
```

Comment:
na początku można zrobić `const { selectedCountry } = this.state

Reply:
Done.